src/Compression/Compression.cs

```diff
@@ -29,12 +32,41 @@ public partial class CompressionStream : Stream
 	{
 		private const int DefaultBufferSize = 8192;
 
-		private Stream _stream;
+		private Stream? _stream;
+		Stream Stream {
+			get {
+				if (_stream is null)
+					throw new ObjectDisposedException (null, "Can not access a closed Stream.");
+				return _stream;
+			}
+		}
 		private CompressionMode _mode;
 		private bool _leaveOpen;
-		private Inflater _inflater;
-		private Deflater _deflater;
-		private byte[] _buffer;
+		private Inflater? _inflater;
+		Inflater Inflater {
+			get {
+				if (_inflater is null)
+					_inflater = new Inflater (_algorithm);
```

Comment:
It looks more appropriate to throw an exception here:

```suggestion
					throw new InvalidOperationException ("Can't access an inflater when decompressing");
```

Reply:
I like the idea, since we should have init the inflater in the constructor, and we can get the nullability check or throw. Using this + the property trick makes the clone simpler to understand.